pkg/api/datasources.go

```diff
@@ -814,10 +813,7 @@ func (hs *HTTPServer) CheckDatasourceHealth(c *contextmodel.ReqContext) response
 func (hs *HTTPServer) checkDatasourceHealth(c *contextmodel.ReqContext, ds *datasources.DataSource) response.Response {
 	pCtx, err := hs.pluginContextProvider.GetWithDataSource(c.Req.Context(), ds.Type, c.SignedInUser, ds)
 	if err != nil {
-		if errors.Is(err, plugincontext.ErrPluginNotFound) {
-			return response.Error(http.StatusNotFound, "Unable to find datasource plugin", nil)
-		}
-		return response.Error(http.StatusInternalServerError, "Unable to get plugin context", err)
+		return response.ErrOrFallback(http.StatusInternalServerError, "Unable to get plugin context", err)
```

Comment:
Don't we need to check whether the error is `ErrPluginNotRegistered` or not and return a specific error like it was done before?

Reply:
No, because the error messages and http statuses are encoded in errutil errors, see https://github.com/grafana/grafana/blob/main/contribute/backend/errors.md